src/HTSPConnection.cpp

```diff
@@ -115,13 +115,13 @@ std::string CHTSPConnection::GetWebURL ( const char *fmt, ... )
   const Settings &settings = Settings::GetInstance();
 
   // Generate the authentication string (user:pass@)
-  std::string auth = settings.strUsername;
-  if (!(auth.empty() || settings.strPassword.empty()))
-    auth += ":" + settings.strPassword;
+  std::string auth = settings.GetUsername();
+  if (!(auth.empty() || settings.GetPassword().empty()))
+    auth += ":" + settings.GetPassword();
```

Comment:
Now that `Settings` is an actual class with actual methods I was thinking we could factor away some things like this to separate methods, e.g. `GetAuthenticationString()` in this case. This may be an edge case since it's technically not that related to "settings" so it's up to you if you want to do it.